sdk/lib/rtl/priv.c

```diff
@@ -486,3 +486,78 @@ RtlAdjustPrivilege(IN ULONG Privilege,
 
     return STATUS_SUCCESS;
 }
+
+/*
+ * @implemented
+ */
+NTSTATUS
+NTAPI
+RtlRemovePrivileges(IN HANDLE TokenHandle,
+                    IN PULONG PrivilegesToKeep,
+                    IN ULONG PrivilegeCount)
+{
+    DPRINT("RtlRemovePrivileges(%p, %p, %u)\n", TokenHandle, PrivilegesToKeep, PrivilegeCount);
+
+    /* Save privileges that should be keep */
+    UINT64 PrivilegesToKeepBitmap = 0;
+    if (PrivilegeCount)
+    {
+        ULONG ulCount;
+        PULONG pulPrivilegesToKeep = PrivilegesToKeep;
+        for (ulCount = 0; ulCount < PrivilegeCount; ulCount++)
+        {
+            PrivilegesToKeepBitmap |= (1ULL << *pulPrivilegesToKeep);
+            if (*pulPrivilegesToKeep++ > SE_MAX_WELL_KNOWN_PRIVILEGE)
+            {
+                return STATUS_INVALID_PARAMETER;
+            }
+        }
+    }
+
+    /* Get token privileges information */
+    ULONG ulReturnLength;
+    BYTE tpBuffer[sizeof(TOKEN_PRIVILEGES) +
+                  sizeof(LUID_AND_ATTRIBUTES) * (SE_MAX_WELL_KNOWN_PRIVILEGE - SE_MIN_WELL_KNOWN_PRIVILEGE)];
+    NTSTATUS Status = ZwQueryInformationToken(TokenHandle,
+                                              TokenPrivileges,
+                                              tpBuffer,
+                                              sizeof(tpBuffer),
+                                              &ulReturnLength);
+    if (NT_SUCCESS(Status))
+    {
+
+        /* Remove all privileges that we don't need to keep */
+        PTOKEN_PRIVILEGES tp = (PTOKEN_PRIVILEGES)tpBuffer;
+        DWORD i;
+        for (i = 0; i < tp->PrivilegeCount; i++)
+        {
+            UINT64 ulPrivilege = ((UINT64)tp->Privileges[i].Luid.HighPart << 32) | tp->Privileges[i].Luid.LowPart;
+            if (PrivilegesToKeepBitmap & (1ULL << ulPrivilege))
+            {
+                PrivilegesToKeepBitmap &= ~(1ULL << ulPrivilege);
+            }
+            else
+            {
+                tp->Privileges[i].Attributes = SE_PRIVILEGE_REMOVED;
+            }
+        }
+
+        if (PrivilegesToKeepBitmap)
+        {
+            Status = STATUS_NOT_ALL_ASSIGNED;
```

Comment:
`STATUS_NOT_ALL_ASSIGNED` is a success code. So it should probably still remove the privileges.

Reply:
Here keeps the same with Windows, in this code path, `RtlRemovePrivileges` returns `STATUS_NOT_ALL_ASSIGNED` but do nothing actually.